src/components/AuthRoute.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { useState, useEffect } from "react";
+import { getAuth, onAuthStateChanged } from "firebase/auth";
+import { useNavigate } from "react-router-dom";
+export interface IAuthRouteProps { }
+
+
+
+const AuthRoute: React.FunctionComponent<IAuthRouteProps> = (props) => {
```

Comment:
**praise:** I like having the type information up front, so the props are typed and the function will not return something other than JSX accidentally 🙂 

**suggestion (non-blocking):** I use `React.FC` instead of `React.FunctionComponent` just to save some characters and have less visual clutter in the type definition.